pfej00-sp2

pfej00 opened the pull request:
Link: https://investin-app.herokuapp.com/, tohle by měla být oprava, původní branch jsem pro jistotu smazal.

pfej00 committed:
pfej00-sp2

pfej00 committed:
pfej00-sp2

Tohle by měla být finální verze aplikace před obhajobou. Prosím o trpělivost při používání aplikace, protože api mi povoluje jenom 5 requestů za minutu.

Link: https://investin-app.herokuapp.com/

nvbach91 commented on www/pfej00/sp2/src/Components/Cryptos.js:
```diff
@@ -0,0 +1,79 @@
+    event.preventDefault();    
+    this.setState({ search: event.target.value });
+    const filteredCryptocurrencies = cryptoList.filter(crypto => {
+            return crypto.name.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1;
```
pouzivejte metodu `includes`

nvbach91 commented on www/pfej00/sp2/src/Components/Cryptos.js:
```diff
@@ -0,0 +1,79 @@
+    for (let index = 0; index < array.length; index++) {
+      let instrument = array[index];
+      instrument = Object.values(instrument);
+      let formulation = <button type="button" className="list-group-item list-group-item-action" onClick = {(e) => {this.saveValueToStorage(e, index)}} >{instrument[0]}</button>;
```
chybi atribut `key`

nvbach91 commented on www/pfej00/sp2/src/Components/DataError.js:
```diff
@@ -0,0 +1,13 @@
+    render() {
+      return (
+        <div className="container">
+            <div class="alert alert-danger" role="alert">{this.props.errorMessage}</div>
```
className

nvbach91 commented on www/pfej00/sp2/src/Components/Stocks.js:
```diff
@@ -0,0 +1,122 @@
+    for (let index = 0; index < this.state.stockOffer.length; index++) {
+      let instrument = this.state.stockOffer[index];
+      //let formulation = <li key={index}><a onClick = {(e) => {this.saveValueToStorage(e, index)}} >{instrument[1]} - {instrument[2]}</a></li>;
+      let formulation = <button type="button" className="list-group-item list-group-item-action" onClick = {(e) => {this.saveValueToStorage(e, index)}} >{instrument[1]} - {instrument[2]}</button>;
```
chybi key prop

nvbach91 reviewed (changes requested):
- odevzdavejte pouze zdrojove soubory, ne build
- moc to nefunguje, ze by chyba na API?

pfej00 committed:
pfej00-sp2 obhajoba

Link: investin-app.herokuapp.com/ finální verze podle zadání na obhajobě

nvbach91 reviewed (changes requested):
- vubec to nefunguje, budu potrebovat abyste prisel znovu
- smazte veskere soubory ve slozce build v tomto pull requestu

pfej00 commented:
Dobrý den,
 promiňte za zpožděnou zpětnou vazbu, protože jsem teď měl několik zkoušek. Mohu provést nový pull request nebo se domluvíme na další obhajobě, případně jaký termín?
 S pozdravem
Jan Pfeiffer, pfej00

pfej00 committed:
pfej00-sp2 Oprava chyb po obhajobě

Link: https://investin-app.herokuapp.com

Opraveny chyby s key prop, includes, className. Navíc jsem našel chybu kvůli který padala aplikace. Chyba nastala v případě kdy uživatel vyhledal akcii tesly na TL0.DEX - Tesla. Tato akcie je speciální v tom, že má v tickeru tečku, což nemívají a když uživatel poslal request s tímto tickerem, tak api nebylo schopný tuto akcii najít. Dále jsem tam přidal další api klíče, takže už by provoz aplikace měl být plynulý.

pfej00 committed:
delete build

Mazu složku build, jak bylo požadováno

pfej00 committed:
pfej00-sp2 final

https://investin-app.herokuapp.com/

nvbach91 commented:
![how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638](https://user-images.githubusercontent.com/20724910/104008962-8b66ea80-51aa-11eb-9b47-d64347f9797a.jpg)

nvbach91 reviewed (approved):
![how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638](https://user-images.githubusercontent.com/20724910/104008962-8b66ea80-51aa-11eb-9b47-d64347f9797a.jpg)
